reactioncommerce/reaction
# feat: GraphQL catalog items

Resolves #4105   
Impact: **minor**  
Type: **feature**

## Changes
This implements the Catalog-related GraphQL schemas and resolvers. There was much discussion around many of the fields during development, so this does not exactly match the proposed schema in the related issue.

There are many tricks in here to map between what is in the Catalog collection and the expected GraphQL schema. There are separate issues to go back and adjust the collection data and related Meteor code, to make those two schemas match much more closely.

## Breaking changes
None

## Testing
1. Verify that you can query for all of the data exposed by the `catalogItems` root GraphQL query, and that the schema of said data makes sense for consumption by frontend clients.
1. Verify that products and variants that are deleted or invisible are not included in the response.
